pandas/core/indexes/interval.py

```diff
@@ -1086,8 +1086,52 @@ def equals(self, other):
                 self.right.equals(other.right) and
                 self.closed == other.closed)
 
-    @Appender(_interval_shared_docs['overlaps'] % _index_doc_kwargs)
```

Comment:
I think there should be a way of reusing the main portion of the docstring, and appending in custom examples.  Something similar to the constructor maybe?

https://github.com/pandas-dev/pandas/blob/bb0376c209d97e49b9c94038926ee62ddc0ee84d/pandas/core/indexes/interval.py#L100-L124